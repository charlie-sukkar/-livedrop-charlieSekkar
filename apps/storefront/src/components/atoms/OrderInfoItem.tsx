import React from 'react';

interface OrderInfoItemProps {
  label: string;
  value: React.ReactNode;
  className?: string;
}

export const OrderInfoItem: React.FC<OrderInfoItemProps> = ({ 
  label, 
  value, 
  className = '' 
}) => {
  return (
    <div className={`flex flex-col sm:flex-row sm:items-center gap-2 ${className}`}>
      <dt className="text-sm font-medium text-gray-500 min-w-[120px] flex-shrink-0">
        {label}
      </dt>
      <dd className="text-sm text-gray-900 flex-1">
        {value}
      </dd>
    </div>
  );
};