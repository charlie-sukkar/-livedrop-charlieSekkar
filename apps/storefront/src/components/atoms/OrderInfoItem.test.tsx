import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import { OrderInfoItem } from './OrderInfoItem';

describe('OrderInfoItem', () => {
  it('renders label and value', () => {
    render(<OrderInfoItem label="Order ID" value="ORD-123" />);
    expect(screen.getByText('Order ID')).toBeInTheDocument();
    expect(screen.getByText('ORD-123')).toBeInTheDocument();
  });

  it('handles different value types', () => {
    render(<OrderInfoItem label="Quantity" value={5} />);
    expect(screen.getByText('5')).toBeInTheDocument();
  });

  it('renders React nodes', () => {
    const value = <span data-testid="badge">Completed</span>;
    render(<OrderInfoItem label="Status" value={value} />);
    expect(screen.getByTestId('badge')).toBeInTheDocument();
  });

  it('applies custom className', () => {
    render(<OrderInfoItem label="Total" value="$100" className="mt-4" />);
    const container = screen.getByText('Total').closest('div');
    expect(container).toHaveClass('mt-4');
  });

  it('has correct styling', () => {
    render(<OrderInfoItem label="Label" value="Value" />);
    
    const label = screen.getByText('Label');
    const value = screen.getByText('Value');
    
    expect(label).toHaveClass('text-gray-500', 'min-w-[120px]');
    expect(value).toHaveClass('text-gray-900', 'flex-1');
  });
});