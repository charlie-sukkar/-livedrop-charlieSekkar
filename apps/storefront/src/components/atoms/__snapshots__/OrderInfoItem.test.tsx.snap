// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`OrderInfoItem Component > matches snapshot with React node value 1`] = `
<div
  class="flex flex-col sm:flex-row sm:items-center gap-2 "
>
  <dt
    class="text-sm font-medium text-gray-500 min-w-[120px] flex-shrink-0"
  >
    Status
  </dt>
  <dd
    class="text-sm text-gray-900 flex-1"
  >
    <span
      class="badge badge-success"
    >
      Completed
    </span>
  </dd>
</div>
`;

exports[`OrderInfoItem Component > matches snapshot with custom className 1`] = `
<div
  class="flex flex-col sm:flex-row sm:items-center gap-2 border-t pt-2"
>
  <dt
    class="text-sm font-medium text-gray-500 min-w-[120px] flex-shrink-0"
  >
    Total
  </dt>
  <dd
    class="text-sm text-gray-900 flex-1"
  >
    $199.99
  </dd>
</div>
`;

exports[`OrderInfoItem Component > matches snapshot with string value 1`] = `
<div
  class="flex flex-col sm:flex-row sm:items-center gap-2 "
>
  <dt
    class="text-sm font-medium text-gray-500 min-w-[120px] flex-shrink-0"
  >
    Order ID
  </dt>
  <dd
    class="text-sm text-gray-900 flex-1"
  >
    ORD-123456
  </dd>
</div>
`;

exports[`OrderInfoItem Component > renders consistently with React node value 1`] = `
<div
  class="flex flex-col sm:flex-row sm:items-center gap-2 "
>
  <dt
    class="text-sm font-medium text-gray-500 min-w-[120px] flex-shrink-0"
  >
    Status
  </dt>
  <dd
    class="text-sm text-gray-900 flex-1"
  >
    <span
      class="badge badge-success"
    >
      Completed
    </span>
  </dd>
</div>
`;

exports[`OrderInfoItem Component > renders consistently with custom className 1`] = `
<div
  class="flex flex-col sm:flex-row sm:items-center gap-2 border-t pt-2"
>
  <dt
    class="text-sm font-medium text-gray-500 min-w-[120px] flex-shrink-0"
  >
    Total
  </dt>
  <dd
    class="text-sm text-gray-900 flex-1"
  >
    $199.99
  </dd>
</div>
`;

exports[`OrderInfoItem Component > renders consistently with string value 1`] = `
<div
  class="flex flex-col sm:flex-row sm:items-center gap-2 "
>
  <dt
    class="text-sm font-medium text-gray-500 min-w-[120px] flex-shrink-0"
  >
    Order ID
  </dt>
  <dd
    class="text-sm text-gray-900 flex-1"
  >
    ORD-123456
  </dd>
</div>
`;
